packages/actor-init-sparql/lib/ActorInitSparql.ts

```diff
@@ -25,57 +25,128 @@ export {
+    comunica-sparql http://fragments.dbpedia.org/2016-04/en [-q] '{ hero { name friends { name } } }' -i graphql
+    comunica-sparql http://fragments.dbpedia.org/2016-04/en [-f] query.sparql'
+    comunica-sparql http://fragments.dbpedia.org/2016-04/en https://query.wikidata.org/sparql ...
+    comunica-sparql hypermedia@http://fragments.dbpedia.org/2016-04/en sparql@https://query.wikidata.org/sparql ...
+
+  Options:
+    -q            evaluate the given SPARQL query string
+    -f            evaluate the SPARQL query in the given file
+    -c            use the given JSON configuration file (e.g., config.json)
+    -t            the MIME type of the output (e.g., application/json)
+    -i            the query input format (e.g., graphql, defaults to sparql)
+    -b            base IRI for the query (e.g., http://example.org/)
+    -l            sets the log level (e.g., debug, info, warn, ... defaults to warn)
+    -d            sets a datetime for querying Memento-enabled archives
+    -p            delegates all HTTP traffic through the given proxy (e.g. http://myproxy.org/?uri=)
+    --lenient     if failing requests and parsing errors should be logged instead of causing a hard crash
+    --help        print this help message
+    --listformats prints the supported MIME types
+    --version     prints version information
+  `);
+
+  public static getScriptOutput(command: string, fallback: string): Promise<string> {
+    return new Promise((resolve, reject) => {
+      exec(command, (error, stdout, stderr) => {
+        if (error) {
+          resolve(fallback);
+        }
+        resolve((stdout || stderr).trimEnd());
+      });
+    });
   }
 
-  public async run(action: IActionInit): Promise<IActorOutputInit> {
-    const args = minimist(action.argv);
-    if (!args.listformats && (!this.queryString && (!(args.q || args.f) && args._.length < (args.c ? 1 : 2) ||
-        args._.length < (args.c ? 0 : 1) || args.h || args.help || args.v || args.version))) {
-      // Print version information
-      if (args.v || args.version) {
-        const comunicaVersion: string = require('../package.json').version;
-        const dev: string = this.isDevelopmentEnvironment() ? '(dev)' : '';
-        const nodeVersion: string = process.version;
-        const npmVersion: string = await this.getScriptOutput('npm -v', '_NPM is unavailable_');
-        const yarnVersion: string = await this.getScriptOutput('yarn -v', '_Yarn is unavailable_');
-        const os = `${OS.platform()} (${OS.type()} ${OS.release()})`;
-        return { stderr: require('streamify-string')(`| software            | version
+  public static isDevelopmentEnvironment(): boolean {
+    return existsSync(`${__dirname}/../test`);
+  }
+
+  public static async buildContext(args: minimist.ParsedArgs, queryOrHttp: number, helpMessage: string,
+    queryString?: string): Promise<any> {
+    const query: boolean = queryOrHttp === 0;
```

Comment:
`queryOrHttp` is used nowhere except here. I would remove the param `queryOrHttp` in favour of a boolean `queryContext`, which would do the same as your current `query` variable.